### PR TITLE
[Relay/TOPI][Frontend] Add tile and repeat operators in Relay and TOPI

### PR DESCRIPTION
**Pull request by Laurawly**
tile and repeat are used both in mxnet and numpy. BlockGrad in mxnet during inference is just like drop out which we skip.

### CODE REVIEW

**Comment by Laurawly** [comment]
@tqchen Why does CI failed at tests/python/contrib/test_sort.py?

File: python/tvm/relay/op/transform.py [hunk 1]
```diff
@@ -316,6 +316,52 @@ def stack(data, axis):
     return _make.stack(data, axis)
 
 
+def repeat(data, repeats, axis):
+    """Repeats elements of an array.
+    By default, repeat flattens the input array into 1-D and then repeats the elements.
```
Comment (yzhliu) [inline python/tvm/relay/op/transform.py]: can we add an example? I guess we can simply copy from numpy's doc.

File: python/tvm/relay/op/transform.py [hunk 2]
```diff
@@ -316,6 +316,52 @@ def stack(data, axis):
...
+    Returns
+    -------
+    ret : relay.Expr
+        The computed result.
+    """
+    return _make.repeat(data, repeats, axis)
+
+
+def tile(data, reps):
+    """Repeats the whole array multiple times.
```
Comment (yzhliu) [inline python/tvm/relay/op/transform.py]: same here

File: python/tvm/relay/op/transform.py [hunk 3]
```diff
@@ -316,6 +316,52 @@ def stack(data, axis):
...
+    data : relay.Expr
+        The input data to the operator.
+
+    reps : tuple of int
+        The number of times repeating the tensor a.
+
+    .. note::
+        Each dim size of reps must be a positive integer. If reps has length d,
+        the result will have dimension of max(d, a.ndim); If a.ndim < d, a is
+        promoted to be d-dimensional by prepending new axes. If a.ndim ? d, reps
```
Comment (yzhliu) [inline python/tvm/relay/op/transform.py]:
```suggestion
        promoted to be d-dimensional by prepending new axes. If data.ndim >= d, reps
```

File: python/tvm/relay/op/transform.py [hunk 4]
```diff
@@ -316,6 +316,52 @@ def stack(data, axis):
...
+def tile(data, reps):
+    """Repeats the whole array multiple times.
+
+    Parameters
+    ----------
+    data : relay.Expr
+        The input data to the operator.
+
+    reps : tuple of int
+        The number of times repeating the tensor a.
```
Comment (yzhliu) [inline python/tvm/relay/op/transform.py]:
```suggestion
        The number of times repeating the tensor data.
```

File: python/tvm/relay/op/transform.py [hunk 5]
```diff
@@ -316,6 +316,52 @@ def stack(data, axis):
...
+
+    .. note::
+        Each dim size of reps must be a positive integer. If reps has length d,
+        the result will have dimension of max(d, a.ndim); If a.ndim < d, a is
+        promoted to be d-dimensional by prepending new axes. If a.ndim ? d, reps
+        is promoted to a.ndim by pre-pending 1's to it.
```
Comment (yzhliu) [inline python/tvm/relay/op/transform.py]:
```suggestion
        is promoted to data.ndim by prepending 1's to reps.
```

File: topi/include/topi/transform.h [hunk 6]
```diff
@@ -719,6 +719,115 @@ inline Tensor where(const Tensor& condition,
...
+* \param tag The tag to mark the operation
+*
+* \return A Tensor whose op member is the repeat operation
+*/
+inline Tensor repeat(const Tensor& x,
+                     int repeats,
+                     int axis,
+                     std::string name = "tensor",
+                     std::string tag = kBroadcast) {
+  int ndim = static_cast<int>(x->shape.size());
```
Comment (yzhliu) [inline topi/include/topi/transform.h]: use `size_t`

File: src/relay/op/tensor/transform.cc [hunk 7]
```diff
@@ -1035,6 +1035,175 @@ RELAY_REGISTER_OP("arange")
...
+  CHECK_EQ(types.size(), 2);
+  const auto* data = types[0].as<TensorTypeNode>();
+  if (data == nullptr) {
+    CHECK(types[0].as<IncompleteTypeNode>())
+        << "tile: expect input type to be TensorType but get "
+        << types[0];
+    return false;
+  }
+  const auto* param = attrs.as<TileAttrs>();
+  const size_t ndim = static_cast<size_t>(data->shape.size());
```
Comment (yzhliu) [inline src/relay/op/tensor/transform.cc]: `data->shape.size()` is already `size_t`, right?, no need to cast.

File: topi/include/topi/transform.h [hunk 8]
```diff
@@ -719,6 +719,115 @@ inline Tensor where(const Tensor& condition,
...
+*
+* \return A Tensor whose op member is the tile operation
+*/
+inline Tensor tile(const Tensor& x,
+                   Array<Integer> reps,
+                   std::string name = "tensor",
+                   std::string tag = kBroadcast) {
+  int ndim = static_cast<int>(x->shape.size());
+  int rdim = static_cast<int>(reps.size());
+  int tdim = (ndim > rdim) ? ndim : rdim;
```
Comment (yzhliu) [inline topi/include/topi/transform.h]: `Array.size()` is size_t, no need to cast to int.

File: src/relay/op/tensor/transform.cc [hunk 9]
```diff
@@ -1035,6 +1035,175 @@ RELAY_REGISTER_OP("arange")
...
+  const Array<Integer>& reps = param->reps;
+  // check dimension match
+  CHECK(!reps.defined())
+    << "repetition array is not defined. data.ndim = " << ndim;
+  const size_t rndim = static_cast<size_t>(reps.size());
```
Comment (yzhliu) [inline src/relay/op/tensor/transform.cc]: delete cast

**Pull request merged**